source/Subsystem/Renderer/TextureLibrary.cpp

```diff
@@ -12,6 +12,16 @@ TextureLibrary::~TextureLibrary()
 
 void TextureLibrary::CompileTexture(Texture& aTexture)
 {
+	for (const Texture& texture : myTextures)
+	{
+		if (texture.mySource == aTexture.mySource)
+		{
+			free(aTexture.mySource);
```

Comment:
Keep in mind that free doesn't call a destructor and is slower than delete. Use delete if possible.